### PR TITLE
go/vt/mysqlctl: optionally store decompressor cmd in manifest

### LINKED ISSUE #12631: Feature Request: store --external-decompressor in backup manifest
### Feature Description

Store `--external-decompressor` in the backup manifest, and use that to decompress a backup compressed with an `external` compressor, unless `--external-decompressor` is provided, in which case use that.

### Use Case(s)

Without this feature, rolling back from an external compressor to a built-in compressor requires a multi-phase approach:

 1. Take a full backup, setting `--compression-engine` to a built-in engine, and also remove the `--external-compressor` flag.
 2. Take another full backup, and this time remove `--external-decompressor`.

Storing `--external-decompressor` would enable a single-step rollback:

 1. Take a full backup, setting `compression-engine` to a builtin and removing `--external-decompressor` and `--external-decompressor`.

`vtbackup` will see that `--external-decompressor` is not set, look for a value in the backup manifest, and attempt to use that.

### PR DESCRIPTION
**Pull request by maxenglander**
## Description

Add a new builtin/xtrabackup flag `--manifest-external-decompressor`. When set the value of that flag is stored in the manifest field `ExternalDecompressor`. This manifest field may be consulted when decompressing a backup that was compressed with an external command.

This feature enables the following flow:

 1. Take a backup using an external compressor
    ```
     vtbackup --compression-engine=external \
              --external-compressor=zstd \
              --manifest-external-decompressor="zstd -d"
    ```
 2. Restore that backup with a mere `Restore` command, without having to specify `--external-decompressor`.

## Related Issue(s)

Fixes #12631.

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation:
     - https://github.com/vitessio/website/pull/1408
     - https://github.com/vitessio/website/pull/1445

### CODE REVIEW

**Comment by maxenglander** [comment]
Note that the `xb_backup` tests took a few retries to pass. It's possible the changes in this PR have made them more brittle...

**Comment by maxenglander** [comment]
I don't think this test failure is related to my changes:

```
--- FAIL: TestVStreamCopyResume (1.02s)
    vstream_test.go:487: 
        	Error Trace:	/home/runner/work/vitess/vitess/go/vt/vtgate/endtoend/vstream_test.go:487
        	Error:      	Expect "type:VGTID vgtid:{shard_gtids:{keyspace:"ks" shard:"-80" gtid:"MySQL56/0784cd3d-c2c3-11ed-86d1-00224850776e:1-133" table_p_ks:{table_name:"t1_copy_resume" lastpk:{fields:{name:"id1" type:INT64} rows:{lengths:1 values:"4"}}}} shard_gtids:{keyspace:"ks" shard:"80-" gtid:"MySQL56/0784cd3d-c2c3-11ed-86d1-00224850776e:1-133"}} keyspace:"ks" shard:"80-"" to match "type:VGTID vgtid:{shard_gtids:{keyspace:"ks" shard:"-80" gtid:".+" table_p_ks:{table_name:"t1_copy_resume" lastpk:{fields:{name:"id1" type:INT64} rows:{lengths:1 values:"[0-9]"}}}} shard_gtids:{keyspace:"ks" shard:"80-" gtid:".+" table_p_ks:{table_name:"t1_copy_resume" lastpk:{fields:{name:"id1" type:INT64} rows:{lengths:1 values:"[0-9]"}}}}} keyspace:"ks" shard:"(80-|-80)""
```

File: go/test/endtoend/backup/xtrabackup/xtrabackup_test.go
Comment (rsajwani) [inline go/test/endtoend/backup/xtrabackup/xtrabackup_test.go:53]: "TestReplicaBackup" is chosen among other test to save test execution time. You can randomly choose other tests (if you want). This will help to give more coverage.

You can ignore this and keep them as you have done right now.

**Review by rsajwani — APPROVED** [review]
LGTM

**Comment by maxenglander** [comment]
Hey @deepthi my intention here was always for it to be a CLI flag in order to support vtbackup-based flows. I'm open to reworking this to be a `Backup` command flag if you think that's better.

File: go/flags/endtoend/vtctld.txt
Comment (deepthi) [inline go/flags/endtoend/vtctld.txt:71]: Why do we need to pass _any_ compression flags to vtctld? vttablet / vtbackup should be sufficient.
@rsajwani we may need to remove older compression related flags too.

Comment (maxenglander) [reply go/flags/endtoend/vtctld.txt:71]: I don't need to! I'm just doing that because CI fails if I don't, because there's pre-existing code that registers `mysqlctl` flags with vtctld.

Do you want me to address this in this PR, or leave it to a separate PR to tackle that?

Comment (rsajwani) [reply go/flags/endtoend/vtctld.txt:71]: @maxenglander , I would recommend to do it in this PR.. Most of the time if we leave it then it will never be address it... 

Also you might need to update the flags here in website 
https://vitess.io/docs/16.0/reference/programs/vtctld/
In general we should update changes related to any program listed here
https://vitess.io/docs/16.0/reference/programs

**Review by deepthi — APPROVED** [review]

**Comment by deepthi** [comment]
@maxenglander we can merge this once the conflicts are resolved.

**Comment by maxenglander** [comment]
@deepthi thanks, done! I also added some missing things to release notes, so if you like please check that, thanks.

**Pull request merged**
